### ISSUE TITLE
`ASYNC100` should not trigger for blocks containing a `yield` statement

### ISSUE THREAD

**Issue by Zac-HD** [python-trio/flake8-async#224]
```python
@contextmanager
def fail_at_with_error(deadline, error):
    with trio.move_on_at(deadline) as cancel_scope:  # noqa: ASYNC100
        yield cancel_scope
    if cancel_scope.cancelled_caught:
        raise error
```

As of the current version, the `noqa` comment is suppressing an error.  However, the `move_on_at()` _could_ be triggered by checkpoints syntactically outside this block, so I think we should avoid false alarms and allow this case.
